src/compiler/checker.ts

```diff
@@ -16477,8 +16510,10 @@ namespace ts {
             const root = getReferenceRoot(node);
             const parent = root.parent;
             const isLengthPushOrUnshift = parent.kind === SyntaxKind.PropertyAccessExpression && (
-                (<PropertyAccessExpression>parent).name.escapedText === "length" ||
-                parent.parent.kind === SyntaxKind.CallExpression && isPushOrUnshiftIdentifier((<PropertyAccessExpression>parent).name));
+                (<PropertyAccessExpression>parent).name.escapedText === "length" || (
+                    parent.parent.kind === SyntaxKind.CallExpression
+                    && isIdentifier((parent as PropertyAccessExpression).name)

```

Comment:
re using the same type:
- our types are correlated with syntax productions and the syntax production in the spec is PrivateIdentifier. For TS API consumers, I think it's convenient if the types look similar to what's in the spec

Re using separate types but casting:
- I am weak and scare easily. I like knowing the checker has my back.
- The types are similar today, but may not be in the future. Either Identifier or PrivateIdentifier could grow new powers, in which case it will be a pain to go back and update the casts

That said, if you recommend a a cast near here, I can add it, but am not sure exactly where to put it.